webapp/channels/src/components/admin_console/admin_definition.tsx

```diff
@@ -0,0 +1,7274 @@
+import CustomURLSchemesSetting from './custom_url_schemes_setting';
+import DataRetentionSettings from './data_retention_settings';
+import CustomDataRetentionForm from './data_retention_settings/custom_policy_form';
+import GlobalDataRetentionForm from './data_retention_settings/global_policy_form';
+import DatabaseSettings from './database_settings.jsx';
+import ElasticSearchSettings from './elasticsearch_settings';
+import {
+    LDAPFeatureDiscovery,
+    SAMLFeatureDiscovery,
+    OpenIDFeatureDiscovery,
+    OpenIDCustomFeatureDiscovery,
+    AnnouncementBannerFeatureDiscovery,
+    ComplianceExportFeatureDiscovery,
+    CustomTermsOfServiceFeatureDiscovery,
+    DataRetentionFeatureDiscovery,
+    GuestAccessFeatureDiscovery,
+    SystemRolesFeatureDiscovery,
+    GroupsFeatureDiscovery,
+} from './feature_discovery/features';
+import FeatureFlags from './feature_flags';
+import GroupDetails from './group_settings/group_details';
+import GroupSettings from './group_settings/group_settings';
+import LicenseSettings from './license_settings';
+import MessageExportSettings from './message_export_settings.jsx';
+import OpenIdConvert from './openid_convert';
+import PasswordSettings from './password_settings';
+import PermissionSchemesSettings from './permission_schemes_settings';
+import PermissionSystemSchemeSettings from './permission_schemes_settings/permission_system_scheme_settings';
+import PermissionTeamSchemeSettings from './permission_schemes_settings/permission_team_scheme_settings';
+import PushNotificationsSettings from './push_settings';
+import ServerLogs from './server_logs';
+import SessionLengthSettings from './session_length_settings';
+import SystemRoles from './system_roles';
+import SystemRole from './system_roles/system_role';
+import SystemUserDetail from './system_user_detail';
+import SystemUsers from './system_users';
+import ChannelSettings from './team_channel_settings/channel';
+import ChannelDetails from './team_channel_settings/channel/details';
+import TeamSettings from './team_channel_settings/team';
+import TeamDetails from './team_channel_settings/team/details';
+import ValidationResult from './validation';
+import WorkspaceOptimizationDashboard from './workspace-optimization/dashboard';
+
+const FILE_STORAGE_DRIVER_LOCAL = 'local';
+const FILE_STORAGE_DRIVER_S3 = 'amazons3';
+const MEBIBYTE = Math.pow(1024, 2);
+
+const SAML_SETTINGS_SIGNATURE_ALGORITHM_SHA1 = 'RSAwithSHA1';
+const SAML_SETTINGS_SIGNATURE_ALGORITHM_SHA256 = 'RSAwithSHA256';
+const SAML_SETTINGS_SIGNATURE_ALGORITHM_SHA512 = 'RSAwithSHA512';
+
+const SAML_SETTINGS_CANONICAL_ALGORITHM_C14N = 'Canonical1.0';
+const SAML_SETTINGS_CANONICAL_ALGORITHM_C14N11 = 'Canonical1.1';
+
+type Component = any
+
+type AdminDefinitionConfigSchemaComponent = {
+    id: string;
+    component: Component;
+}
+
+type ConsoleAccess = {read: any; write: any}
+
+type Validator = (value: any) => ValidationResult
+
+type AdminDefinitionSettingCustom = {
```

Comment:
Could you move these type definitions into a separate file like `components/admin_console/types.tsx` to avoid adding more lines to this file? This thing's already pretty massive

Reply:
Totally agree.

Reply:
Done